packages/frontend/src/components/common/ShareSpaceModal/ShareSpaceUserList.tsx

```diff
@@ -296,15 +333,43 @@ type GroupAccessListProps = {
         action: UserAccessAction,
         currentGroupAccess: SpaceGroup,
     ) => void;
+    pageSize?: number;
 };
 const GroupsAccessList: FC<GroupAccessListProps> = ({
     isPrivate,
     onAccessChange,
     groupsAccess,
+    pageSize,
 }) => {
+    const [page, setPage] = useState(1);
+
+    // TODO: Paginate group access from backend
+    const paginatedList: SpaceGroup[][] = useMemo(() => {
+        const list: SpaceGroup[][] = [];
+        const accessListClone = cloneDeep(groupsAccess);
```

Comment:
why is this necessary?

Reply:
fyi, if it's absolutely necessary: https://www.builder.io/blog/structured-clone

Reply:
This is necessary because we're using splice which mutates the array.
Changed for `structuredClone` 👌